test/gurka/test_gtfs.cc

```diff
@@ -0,0 +1,137 @@
+    .route_text_color = "black",
+  };
+  feed.add_route(lineOne);
+  feed.write_routes("../test/data/gtfs_test/");
+
+  // write trips.txt
+  struct gtfs::Trip tripOne {
+    .route_id = "baz", .service_id = "bon", .trip_id = "bar", .trip_headsign = "haha",
+    .block_id = "block", .shape_id = "square", .wheelchair_accessible = gtfs::TripAccess::Yes,
+    .bikes_allowed = gtfs::TripAccess::No,
+  };
+  feed.add_trip(tripOne);
+  feed.write_trips("../test/data/gtfs_test/");
+
+  // write stop_times.txt
+  struct StopTime stopTimeOne {
+    .trip_id = "bar", .arrival_time = Time("6:00:00"), .departure_time = Time("6:00:00"),
+    .stop_id = "foo", .stop_sequence = 0, .stop_headsign = "head", .shape_dist_traveled = 0.0,
+    .timepoint = gtfs::StopTimePoint::Exact,
+  };
+  feed.add_stop_time(stopTimeOne);
+  feed.write_stop_times("../test/data/gtfs_test/");
+
+  // write calendar.txt
+  struct CalendarItem calendarOne {
+    .service_id = "bon", // connected to trip
+        .monday = CalendarAvailability::Available, .tuesday = CalendarAvailability::Available,
+    .wednesday = CalendarAvailability::Available, .thursday = CalendarAvailability::Available,
+    .friday = CalendarAvailability::Available, .saturday = CalendarAvailability::Available,
+    .sunday = CalendarAvailability::Available, .start_date = Date(2022, 1, 31),
+    .end_date = Date(2022, 2, 1),
+  };
+  feed.add_calendar_item(calendarOne);
+  feed.write_calendar("../test/data/gtfs_test/");
+
+  // write calendar_dates.txt
+  struct CalendarDate dateOne {
+    .service_id = "bon", // connected to trip
+        .date = Date(2022, 2, 2), .exception_type = gtfs::CalendarDateException::Added,
+  };
+  feed.add_calendar_date(dateOne);
+  feed.write_calendar_dates("../test/data/gtfs_test/");
+
+  // write shapes.txt
+  struct ShapePoint shapePointOne {
+    .shape_id = "square", .shape_pt_lat = 0.2, .shape_pt_lon = 0.2, .shape_pt_sequence = 1
+  };
+  struct ShapePoint shapePointTwo {
+    .shape_id = "square", .shape_pt_lat = 0.21, .shape_pt_lon = 0.21, .shape_pt_sequence = 2
+  };
+  struct ShapePoint shapePointThree {
+    .shape_id = "square", .shape_pt_lat = 0.22, .shape_pt_lon = 0.22, .shape_pt_sequence = 3
+  };
+  struct ShapePoint shapePointFour {
+    .shape_id = "square", .shape_pt_lat = 0.23, .shape_pt_lon = 0.23, .shape_pt_sequence = 4
+  };
+  feed.add_shape(shapePointOne);
+  feed.add_shape(shapePointTwo);
+  feed.add_shape(shapePointThree);
+  feed.add_shape(shapePointFour);
+  feed.write_shapes("../test/data/gtfs_test/");
+
+  // write frequencies.txt
+  struct Frequency freqOne {
+    .trip_id = "bar", .start_time = Time(0, 0, 0), .end_time = Time(2, 0, 0), .headway_secs = 1800,
+    .exact_times = gtfs::FrequencyTripService::FrequencyBased,
+  };
+  feed.add_frequency(freqOne);
+  feed.write_frequencies("../test/data/gtfs_test/");
+
+  // make sure files are actually written
+  const auto& trips = feed.get_trips();
```

Comment:
you'll need to read the feed off disk, i.e. create a new feed from the directory path